src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static class Node<T> {
+        private Node<T> next;
+        private T value;
+        private Node<T> prev;
+
+        Node(T value, Node<T> prev, Node<T> next) {
+            this.value = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (tail == null) {
+            this.head = this.tail = new Node(value, null, null);
```

Comment:
You can avoid warnings from Idea (Raw use of parameterized class 'Node' ) if you add '<>' here. Fix it in other places where you have yellow warnings from Idea
```suggestion
            head = tail = new Node<>(value, null, null);
```